android/src/main/kotlin/com/web3auth/flutter/web3auth_flutter/Web3AuthFlutterPlugin.kt

```diff
@@ -78,7 +78,11 @@ class Web3AuthFlutterPlugin : FlutterPlugin, ActivityAware, MethodCallHandler,
                 launch(Dispatchers.Main) { result.notImplemented() }
             } catch (e: Throwable) {
                 launch(Dispatchers.Main) {
-                    result.error("error", e.message, e.localizedMessage)
+                    if(e.message == "User Cancelled") {
+                        result.error("UserCancelledException", e.message, e.localizedMessage)
+                    } else {
+                        result.error("error", e.message, e.localizedMessage)
+                    }
```

Comment:
Can we handle other error as well which we have used _handlePlatformException. Otherwise that error would have same issue, if someone is using that exception, it'll not be catched.

Reply:
As discussed, its a major refactored wrt to handling all errors across Swift and kotlin. Will take this as performance improvement.